### PR TITLE
Lint and other small fixes to make the final TypeScript conversion easier to read

### PR DESCRIPTION
**Pull request by buggy**
*Issue #, if available:*

#1028

*Description of changes:*

@powerful23 While the PR looks huge it should actually be quick to review.

This PR contains many small fixes that typescript or tslint will require for the final conversion to TypeScript. Almost all fall into one of a few categories:

- Missing `;`
- Missing blank line
- Using `var` instead of `let` or `const`
- Using `let` instead of `const`
- Using `{ someVar: someVar }` instead of `{ someVar }`

I've included these in their own PR so that they're easy to review and so the final TypeScript conversion isn't cluttered with minor syntax changes making it easier to review too.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify-react/src/Storage/S3Album.js [hunk 1]
```diff
@@ -175,9 +175,9 @@ export default class S3Album extends Component {
             if (!item.contentType) { item.contentType = this.contentType(item); }
         });
 
-        list = this.filter(list);
-        list = this.sort(list);
-        this.setState({ items: list });
+        let items = this.filter(list);
+        items = this.sort(list);
```
Comment (powerful23) [inline packages/aws-amplify-react/src/Storage/S3Album.js]: Should it be `this.sort(items);`?

**Comment by buggy** [comment]
@powerful23 You're right. I've fixed it.

File: packages/aws-amplify-react/src/Storage/S3Album.js (hunk 1 above)
Comment (buggy) [reply packages/aws-amplify-react/src/Storage/S3Album.js]: You're right. I've fixed it.

**Comment by buggy** [comment]
@kristianmandrup it depends how big the PR is. I may split it into 2 more just to make the reviews easier.

**Pull request merged**